### ISSUE TITLE
Accessing instance, PK, or criteria in lifecycle callbacks

### ISSUE THREAD

**Issue by mmiller42** [balderdashy/sails#5398]
I have a blog system that converts a post title into a slug (e.g. `Hello World` -> `hello-world`). Titles don't need to be unique, but slugs do, so I need to add logic to my model that determines if a post already has the `hello-world` slug and, if so, changes the slug to `hello-world-2` and so on.

Waterline lifecycle callbacks don't receive an instance, only the attributes that are being changed. This is understandable, since the underlying implementation seems to call `Model.update`, not `record.save`.

The issue I'm having is that, if I have just one post called `Hello World` with slug `hello-world`, and I edit and update that post, it changes the slug to `hello-world-2`, because, of course, there already is a post in the database with slug `hello-world`. If I knew the ID of the post I was editing, I could add a condition in my slug conflict checker function that says `if (potentialConflictPost.id !== postWeAreEditing.id`). But without having access to the PK in the lifecycle callback, I have no way of knowing this information.

Basically, I think it's a crucial feature to get the instance, PK, or criteria passed into the `beforeCreate` and `beforeUpdate` lifecycle callbacks somehow.
